src/org/traccar/reports/ReportUtils.java

```diff
@@ -263,10 +263,16 @@ private static StopReport calculateStop(
         stop.setSpentFuel(calculateFuel(startStop, endStop));
 
         long engineHours = 0;
-        for (int i = startIndex + 1; i <= endIndex; i++) {
-            if (positions.get(i).getBoolean(Position.KEY_IGNITION)
-                    && positions.get(i - 1).getBoolean(Position.KEY_IGNITION)) {
-                engineHours += positions.get(i).getFixTime().getTime() - positions.get(i - 1).getFixTime().getTime();
+        if (startStop.getAttributes().containsKey(Position.KEY_HOURS)
+                && endStop.getAttributes().containsKey(Position.KEY_HOURS)) {
+            engineHours = endStop.getLong(Position.KEY_HOURS) - startStop.getLong(Position.KEY_HOURS);
+        } else if (Context.getConfig().getBoolean("engineHours.enable")) {
+            for (int i = startIndex + 1; i <= endIndex; i++) {
```

Comment:
Do we really need this fallback? Or just say that if you want engine hours enable it in the config?

Reply:
This fallback for old data when accumulation was not enabled. Like we did for `totalDistance` (in June 2018 we should remove it if I'm not mistaken)